src/display/text_layer.js

```diff
@@ -57,11 +57,66 @@ const renderTextLayer = (function renderTextLayerClosure() {
 
   const NonWhitespaceRegexp = /\S/;
 
+  const DEFAULT_FONT_SIZE = 30;
+  const DEFAULT_FONT_ASCENT = 0.8;
+
   function isAllWhitespace(str) {
     return !NonWhitespaceRegexp.test(str);
   }
 
-  function appendText(task, geom, styles) {
+  function getAscent(fontFamily, ctx) {
+    const savedFont = ctx.font;
+    ctx.font = `${DEFAULT_FONT_SIZE}px ${fontFamily}`;
+    const metrics = ctx.measureText("");
+
+    // Both properties aren't available by default in Firefox.
+    let ascent = metrics.fontBoundingBoxAscent;
+    let descent = Math.abs(metrics.fontBoundingBoxDescent);
+    if (ascent) {
```

Comment:
As above, please avoid the zero-divisor here.

Reply:
These numbers are a distance so they must be positive or zero:
https://developer.mozilla.org/en-US/docs/Web/API/TextMetrics/actualBoundingBoxRight
so if ascent is non-zero, divisor is non-zero too.